src/index.js

```diff
@@ -50,41 +32,25 @@ Multihashing.multihash = multihash
  * @param {Buffer} buf - The value to hash.
  * @param {number|string} func - The algorithm to use.
  * @param {number} [length] - Optionally trim the result to this length.
- * @param {function(Error, Buffer)} callback
- * @returns {undefined}
+ * @returns {Promise}
  */
-Multihashing.digest = function (buf, func, length, callback) {
-  if (typeof length === 'function') {
-    callback = length
-    length = undefined
-  }
-
-  if (!callback) {
-    throw new Error('Missing callback')
-  }
-
-  let cb = callback
-  if (length) {
-    cb = (err, digest) => {
-      if (err) {
-        return callback(err)
-      }
-
-      callback(null, digest.slice(0, length))
-    }
-  }
-
-  let hash
+Multihashing.digest = (buf, func, length) => {
   try {
-    hash = Multihashing.createHash(func)
+    return Multihashing.createHash(func)(buf)
+      .then(digest => {
+        if (length) {
+          return digest.slice(0, length)
+        }
+        return digest
+      })
   } catch (err) {
-    return cb(err)
+    return Promise.reject(err)
```

Comment:
I don't understand what this catch block would catch. if createHash() fails, it will be an `Unhandled promise rejection` and won't reach the catch case. And if something within the `then()` throws, it also won't reach the catch case.

I needed a minimal example, it might help other reviewers as well to reason about this:

```javascript
'use strict'

const somePromiseResolves = new Promise((resolve) => {
  console.log('`somePromiseResolves` is about to be resolved')
  resolve()
})

const somePromiseRejects = new Promise((_resolve, reject) => {
 console.log('`somePromiseRejects` is about to be rejected')
 reject()
})

const someFunctionResolves = () => {
  try {
    somePromiseResolves.then(
      () => {
        console.log('`somePromiseResolves` was resolved')
        throw new Error('throwing in the success case')
      })
  } catch (error) {
    console.log('`somePromiseResolves` was rejected')
    Promise.reject(error)
  }
}

const someFunctionRejects = () => {
  try {
    somePromiseRejects.then(
      () => {
        console.log('`somePromiseRejected` was resolved')
      })
  } catch (error) {
    console.log('`somePromiseRejected` was rejected')
    Promise.reject(error)
  }
}

someFunctionResolves()
someFunctionRejects()
```

Reply:
@hugomrdias ^^